src/pages/index.tsx

```diff
@@ -53,38 +57,73 @@ const StyledGoalsList = styled.div`
     padding: 20px 20px 0 20px;
 `;
 
+const StyledLoadMore = styled.div`
+    margin: 50px 40px;
+`;
+
 export const getServerSideProps = declareSsrProps(async ({ user }) => ({
     ssrData: await fetcher(user),
 }));
 
 export default declarePage<{ goalUserIndex: Goal[] }>(({ user, locale, ssrData }) => {
     const t = useTranslations('index');
-    const { data } = useSWR('goalUserIndex', () => fetcher(user));
-    const goals = data?.goalUserIndex ?? ssrData.goalUserIndex;
+
+    const { data, setSize, size } = useSWRInfinite(
+        (index: number) => ({ offset: index * PAGE_SIZE }),
+        ({ offset }) => fetcher(user, offset),
+    );
+
+    function shouldRenderMoreButton() {
+        const firstElement = data?.[0];
+        const lastElement = data?.[data.length - 1];
+
+        const isFirstPageEmpty = firstElement?.goalUserIndex?.length === 0;
+        const isLastPageFull =
+            lastElement && lastElement.goalUserIndex && lastElement.goalUserIndex.length === PAGE_SIZE;
+
+        if (isFirstPageEmpty) {
+            return false;
+        }
+
+        if (isLastPageFull) {
+            return true;
+        }
+        return false;
+    }
+
+    let goals: Goal[] = [];
+    if (data) {
+        for (const { goalUserIndex } of data) {
+            if (goalUserIndex) {
+                goals.push(...goalUserIndex);
+            }
+        }
+    } else {
```

Comment:
Instead of this statement look at [this example](https://github.com/taskany-inc/issues/blob/main/src/pages/goals/%5Bid%5D.tsx#L107-L109) of using ssrData as fallback for client data. It can help you manipulate data in one way.
Doc about fallback [here](https://swr.vercel.app/docs/with-nextjs).